### PR TITLE
Benchmark script and performance improvement for augeas plugin

### PR DESCRIPTION
**Pull request by fberlakovich**
The PR contains a small script (should be sh compatible :)) for benchmarking the augeas plugin performance with a provided hosts file. While analysing the plugin with callgrind I realised that an easy performance improvement can be made which is also included.

### CODE REVIEW

File: scripts/benchmark_augeas_hosts.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+for run in $(seq 0 11); do
+	echo "RUN: #$run"
+	augeaswalltime=$(measure_time "$KDB ls $AUGEASPATH")
+	hostswalltime=$(measure_time "$KDB ls $HOSTSPATH")
+	keytometawalltime=$(measure_time "$KDB ls $KEYTOMETAPATH")
+	echo "augeas: $augeaswalltime; hosts: $hostswalltime; augeas with keytometa: $keytometawalltime"
+done
+
+for x in ` kdb mount | grep system/benchmarks/$$ | awk '{print $3}'`; do sudo kdb umount $x; done
```
Comment (markus2330) [inline scripts/benchmark_augeas_hosts.sh]: Its strange that mounting is done without sudo, but umount with. Also $KDB is not used and the whitespaces could be improved.

File: src/plugins/augeas/augeas.c [hunk 2]
```diff
@@ -198,7 +198,7 @@ static int removeOrphan (augeas * handle, const char * treePath, void * data)
 static int foreachAugeasNode (augeas * handle, const char * treePath, ForeachAugNodeClb callback, void * callbackData)
 {
 	char * matchPath;
-	asprintf (&matchPath, "%s/*", treePath);
+	asprintf (&matchPath, "%s//*", treePath);
```
Comment (markus2330) [inline src/plugins/augeas/augeas.c]: Thanks, but this should be a different commit.

File: src/plugins/augeas/augeas.c [hunk 3]
```diff
@@ -216,9 +216,6 @@ static int foreachAugeasNode (augeas * handle, const char * treePath, ForeachAug
 
 		result = (*callback) (handle, curPath, callbackData);
 
-		/* handle the subtree */
-		result = foreachAugeasNode (handle, curPath, callback, callbackData);
```
Comment (markus2330) [inline src/plugins/augeas/augeas.c]: Are you sure that this is right?

**Comment by markus2330** [comment]
Thank you, I added some comments!

File: src/plugins/augeas/augeas.c (hunk 3 above)
Comment (fberlakovich) [inline src/plugins/augeas/augeas.c]: At least the tests say so :). This can be done because ```path//*``` is used instead of ```path/*````. While ```path/*``` returns only the nodes below path, ```path//*``` returns all subtrees below path. See https://github.com/hercules-team/augeas/wiki/Path-expressions for more details.

Comment (markus2330) [inline src/plugins/augeas/augeas.c]: Ok, nice, thank you. Is this also faster or otherwise unrelated to the benchmark?

Comment (fberlakovich) [inline src/plugins/augeas/augeas.c]: It is about 20% faster than the old solution
